jacodb-panda-dynamic/src/main/kotlin/org/jacodb/panda/dynamic/ets/model/EtsSignature.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Copyright 2022 UnitTestBot contributors (utbot.org)
+ * <p>
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ * <p>
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.jacodb.panda.dynamic.ets.model
+
+import org.jacodb.api.common.CommonMethodParameter
+import org.jacodb.panda.dynamic.ets.base.EtsType
+
+data class EtsFileSignature(
+    val projectName: String,
+    val fileName: String,
+) {
+    override fun toString(): String {
+        // Remove ".d.ts" and ".ts" file ext:
+        val tmp = fileName.replace("""(\.d\.ts|\.ts)$""".toRegex(), "")
```

Comment:
This replacement is very expensive. Why don't you use file methods or don't remove it by yourself?

Reply:
This was copied from ArkAnalyzer.

I do not think this operation is *very expensive*. Surely, compiling a Regex is not free, but the regex itself is simple (`$` constrains it heavily, so the matching should be fast), and the class `EtsFileSignature`, in fact, is used almost 0 times, so we can expect that this "costy" `toString` won't be a bottleneck.

Note that the straightforward `fileName.removeSuffix(".d.ts").removeSuffix(".ts")` is semantically different, since it might cut too much if the file is called `kek.ts.d.ts`. You can suggest another solution in replacement for the regex.